src/app/shared/components/label-form/template.html

```diff
@@ -19,56 +19,71 @@
-          Name is too long.
-        </mat-error>
-        <mat-error *ngIf="label.get('key').errors?.labelKeyNameRestricted">
-          Restricted label name.
-        </mat-error>
-      </mat-form-field>
-      <mat-form-field fxFlex="45">
-        <mat-label>{{valueName}}</mat-label>
-        <input matInput
-               (keyup)="check(i)"
-               name="value"
-               formControlName="value">
-        <mat-error *ngIf="label.get('value').errors?.validLabelValuePattern">
-          {{valueName}} not allowed.
-        </mat-error>
-        <mat-error *ngIf="label.get('value').errors?.validLabelValueLength">
-          {{valueName}} is too long.
-        </mat-error>
-      </mat-form-field>
-      <button mat-icon-button
-              type="button"
-              class="delete-button"
-              [disabled]="!isRemovable(i)"
-              (click)="deleteLabel(i)">
-        <i class="km-icon-mask km-icon-delete"
-           aria-hidden="true"></i>
-      </button>
+         fxLayout="column">
+      <div [formGroupName]="i"
+           fxLayout="row"
+           fxLayoutGap="10px">
+        <mat-form-field fxFlex="45">
+          <mat-label>{{keyName}}</mat-label>
+          <input matInput
+                 (keyup)="check(i)"
+                 name="key"
+                 formControlName="key" />
+          <mat-error *ngIf="label.get('key').errors?.validLabelKeyUniqueness"> {{keyName}} is not unique. </mat-error>
+          <mat-error *ngIf="label.get('key').errors?.validLabelKeyPrefixPattern"> Prefix not allowed. </mat-error>
+          <mat-error *ngIf="label.get('key').errors?.validLabelKeyNamePattern"> Name not allowed. </mat-error>
+          <mat-error *ngIf="label.get('key').errors?.validLabelKeyPrefixLength"> Prefix is too long. </mat-error>
+          <mat-error *ngIf="label.get('key').errors?.validLabelKeyNameLength"> Name is too long. </mat-error>
+          <mat-error *ngIf="label.get('key').errors?.labelKeyNameRestricted"> Restricted label name. </mat-error>
+        </mat-form-field>
+        <mat-form-field fxFlex="45">
+          <mat-label>{{valueName}}</mat-label>
+          <input matInput
+                 (keyup)="check(i)"
+                 name="value"
+                 formControlName="value" />
+          <mat-error *ngIf="label.get('value').errors?.validLabelValuePattern"> {{valueName}} not allowed. </mat-error>
+          <mat-error *ngIf="label.get('value').errors?.validLabelValueLength"> {{valueName}} is too long. </mat-error>
+        </mat-form-field>
+        <button mat-icon-button
+                type="button"
+                class="delete-button"
+                [disabled]="!isRemovable(i)"
+                (click)="deleteLabel(i)">
+          <i class="km-icon-mask km-icon-delete"
+             aria-hidden="true"></i>
+        </button>
+      </div>
+
+      <mat-hint *ngIf="(label?.value?.key ?? '').toLowerCase?.() === labelHintKey?.toLowerCase()"
```

Comment:
Is it a requirement that labels should be case insensitive?

Reply:
this is a functionality question not a UX question imo, which way makes more sense?

Reply:
@Waseem826 what's your opinion? should I keep ignoring case sensitivity while matching keys?

Reply:
I think we should ask someone from backend team.

Reply:
@hdurand0710 can you give some input here by chance?

Reply:
Field selectors (including label selectors) are case sensitive. It's important that the node labels matches the label selector regarding the case sensitivity.